server/src/main/java/org/apache/cassandra/sidecar/utils/SidecarClientProvider.java

```diff
@@ -76,6 +77,21 @@ public SidecarClient get()
         return client;
     }
 
+    public void close(Promise<Void> completion)
+    {
+        LOGGER.info("Closing Sidecar Client...");
+        try
+        {
+            client.close();
+            Thread.sleep(100);
+            completion.complete();
+        }
+        catch (Throwable throwable)
+        {
+            completion.fail(throwable);
+        }
+    }
```

Comment:
This is not really async. Let's not declare promise input. See the suggestion below, and at the callsite, i.e. `Server#close`, have this instead. 
```
        Future.future(p -> {
            sidecarClientProvider.close();
            p.complete();
        });
```

```suggestion
     private final AtomicBoolean isClosing = new AtomicBoolean(false);
    public void close()
    {
        if (isClosing.compareAndSet(false, true))
        {
            LOGGER.info("Closing Sidecar Client...");
            ThrowableUtils.propagate(client::close);
        }
    }
```